### DISCUSSION TITLE
[Feature] option to disable timezone correction based on geo coordinates

### DISCUSSION THREAD

**Discussion by aisbergde** [immich-app/immich#6189] [Feature Request]
### The feature

I want an option to disable timezone correction based on geo coordinates at all. I want assets to be listed by their embedded time in their natural order.

My issue: When some assets have geo coordinates, then immich uses the local time at this geo position. When  other assets don't have geo coordinates, it uses my immich user local time. Finally my assets are in the wrong order because of the timezone correction and a mix of assets, containing or not containing geo data. I just want to have assets in the right order, not depending on existing geo data.

My current solution is to use external programs to assign missing geo data. But this is a huge work. And it is not so easy to deside the coordinates.

related to #3863, but I want to suggest a simple solution for a simple issue.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by jenda69** [comment]
So that is the reason my photos and videos are in weird order! Because I was geotagging photos but not videos from my camera.

Well, I tried to add coordinates to videos too and it's still broken and I don't understand that at all.

I have video and photo taken today at 16:19:xx using the same camera. AFAIK there's no timezone info in the metadata. I added GPS coordinates from GPX using Exiftool.

Immich correctly says the photo is taken at 16:19 GMT+01:00, it understands that the embeded time is just missing the time zone info.
But for some reason, the video shows 17:19 GMT+01:00, like if it thought the embeded time is UTC so it adds an hour to the timestamp. WHY??? 🤦‍♂️
edit: Oh, I see it. According to the specification, metadata in video should be in UTC. But that often isn't the case (like for my camera) so exiftool just shows it as local time. But Immich evidently takes the times in videos as UTC... Well I guess I'll have to shift the times in my videos.

**Comment by aisbergde** [comment]
An immich option to turn off timezone correction based on geo location would solve a lot of issues. For now I use third party software like digiKam to maintain geolocation. Regarding videos - geo location needs to be stored into sidecar files using a naming pattern, accepted by immich. filename.extension.xmp could work.

**Reply by jenda69** [reply]
Yeah, sorry I think my issue is off topic here actually. But your remark about sidecars made me do little test with that. And I don't understand that even more now. I generated XMP files from the videofiles using `exiftool -ext mp4 -o %d%f.%e.xmp ./` and after getting Immich to rescan the files with the sidecars, time is actually correct now without changing or shifting them.

**Reply by aisbergde** [reply]
Maybe, you included geo information into the sidecar files, immich extracted them and did the time correction based on this geo meta data? And now the order is corrected?

**Comment by DeltaTango69** [comment]
Ahh, I suspected yesterday that Immich was somehow using the offset Exif data. That's why I have a huge mess in the timeline. See my article. Time offset in timeline UTC - local time #6674. So this is probably really the case. Please change this behaviour. I have over 50000 pictures without GPS data where I no longer know where they were taken. I can't possibly enter that.
